spanner/google/cloud/spanner/streamed.py

```diff
@@ -130,7 +159,35 @@ def consume_next(self):
 
         Parse the result set into new/existing rows in :attr:`_rows`
         """
-        response = six.next(self._response_iterator)
+        sleep_generator = retry.exponential_sleep_generator(
+            initial=retry._DEFAULT_INITIAL_DELAY,
+            maximum=retry._DEFAULT_MAXIMUM_DELAY)
+
+        deadline = time.time() + _RESTART_DEADLINE
+
+        response = None
+
+        for sleep in sleep_generator:
+            try:
+                response = six.next(self._response_iterator)
+            except exceptions.ServiceUnavailable:
+
+                if self._resume_token in (None, b''):
```

Comment:
Seems strange to have to check both of these, why wouldn't `b''` just be the default?

Reply:
`None` is "never set";  `b''` is "the back-end send an empty token."

Reply:
Is there a place where the distinction matters? (It doesn't matter here.)